main.py

```diff
@@ -1680,6 +1706,23 @@ def timer():
                         config["last_weighting_update"] = str(current_milli_time())
                         cur.execute("UPDATE config SET value = %s WHERE name = 'last_weighting_update'",
                                     [config["last_weighting_update"]])
+
+                    # pudding expiry?
+                    now = datetime.datetime.now()
+                    ymdNow = now.strftime("%Y-%m-%d")
+                    if ymdNow != config["last_pudding_check"]:
+                        config["last_pudding_check"] = ymdNow
+                        cur.execute("UPDATE config SET value = %s WHERE name = 'last_pudding_check'", [ymdNow])
```

Comment:
Is there any particular reason you're doing a check based on textual dates, rather than just using the omnipresent `current_milli_time()` and just clamping the timestamp to be accurate to a day?

Reply:
For once I want this code to be run once a day, not once every 24 hours. While I'm not sure if any relevant country can have its DST change over on the 1st or the 8th, it seems worth it to not risk anything.

Reply:
Which reminds me: Can this double-process on leap seconds? I honestly don't know enough about leap seconds to tell, but I'm not sure what result you will get at 23:59:60.